Firmware/planner.cpp

```diff
@@ -229,7 +229,7 @@ void calculate_trapezoid_for_block(block_t *block, float entry_speed, float exit
 #ifdef LIN_ADVANCE
   uint16_t final_adv_steps = 0;
   uint16_t max_adv_steps = 0;
-  if (block->use_advance_lead) {
+  if (block->flag & BLOCK_FLAG_USE_ADVANCE_LEAD) {
```

Comment:
According to the assembly output `if(block->flag & BLOCK_FLAG_USE_ADVANCE_LEAD)` is actually faster than `if(block->use_advance_lead)`

With `if(block->use_advance_lead)` there is a subtraction compare.


```asm
  if (block->use_advance_lead) {
   33cb4:	f1 01       	movw	r30, r2        ; 1 cycle - Copy Register Word
   33cb6:	e4 5b       	subi	r30, 0xB4	   ; 1 cycle - Subtract Immediate
   33cb8:	ff 4f       	sbci	r31, 0xFF      ; 1 cycle - Subtract Immediate with Carry
   33cba:	90 81       	ld	r25, Z             ; 1-3 cycles - Load Indirect From Data Space
   33cbc:	9b 8f       	std	Y+27, r25	       ; 2 cycles - Store Indirect From Register to Data Space
   33cbe:	99 23       	and	r25, r25           ; 1 cycle - Logical AND
   33cc0:	09 f4       	brne	.+2      	; 0x33cc4 <calculate_trapezoid_for_block(block_t*, float, float)+0x1f4>
   33cc2:	8d c0       	rjmp	.+282    	; 0x33dde <calculate_trapezoid_for_block(block_t*, float, float)+0x30e>
      final_adv_steps = final_rate * block->adv_comp;
   33cc4:	61 2f       	mov	r22, r17
```



```asm
  if (block->flag & BLOCK_FLAG_USE_ADVANCE_LEAD) {
   33c68:	b8 2f       	mov	r27, r24        ; 1 cycle - Copy Register
   33c6a:	b0 74       	andi	r27, 0x40	; 1 cycle - Logical AND with Immediate
   33c6c:	bb 8f       	std	Y+27, r27	    ; 2 cycles - Store Indirect From Register to Data Space
   33c6e:	86 ff       	sbrs	r24, 6      ; 1-3 cycles - Skip if Bit in Register is Set
   33c70:	93 c0       	rjmp	.+294    	; 0x33d98 <calculate_trapezoid_for_block(block_t*, float, float)+0x31a>
      final_adv_steps = final_rate * block->adv_comp;
   33c72:	61 2f       	mov	r22, r17
```